### PR TITLE
fix(earn): always ask for companion signature if it does not have per…

### PR DESCRIPTION
**Pull request by FiboApe**
…mission

### CODE REVIEW

File: apps/root/src/services/earnService.ts [hunk 1]
```diff
@@ -527,11 +528,30 @@ export class EarnService extends EventsManager<EarnServiceData> {
...
+        permissions: [EarnPermission.INCREASE],
+      },
+    ];
+
+    const wrappedProtocol = getWrappedProtocolToken(strategy.farm.chainId);
+
+    if (strategy.farm.asset.address === wrappedProtocol.address) {
+      permissions.push({
+        operator: earnCompanionAddress,
+        permissions: [EarnPermission.WITHDRAW],
```
Comment (th0rOdinson) [inline apps/root/src/services/earnService.ts:545]: Why add this on position creation?

File: apps/root/src/pages/strategy-guardian-detail/strategy-management/deposit/hooks/useEarnDepositActions.tsx [hunk 2]
```diff
@@ -603,14 +603,13 @@ const useEarnDepositActions = ({ strategy }: UseEarnDepositActionParams) => {
...
     }
-    const wrappedProtocolToken = getWrappedProtocolToken(strategy.network.chainId);
     const position = strategy.userPositions?.find((userPosition) => userPosition.owner === activeWallet.address);
     const companionHasPermission =
       !isIncrease ||
       position?.permissions[contractService.getEarnCompanionAddress(strategy.network.chainId)]?.includes(
         EarnPermission.INCREASE
       );
-    return wrappedProtocolToken?.address === strategy?.asset.address && isIncrease && !companionHasPermission;
+    return isIncrease && !companionHasPermission;
```
Comment (th0rOdinson) [inline apps/root/src/pages/strategy-guardian-detail/strategy-management/deposit/hooks/useEarnDepositActions.tsx:612]: A create also uses permit2 right? Wouldn't that need companion too?

File: apps/root/src/services/earnService.ts (hunk 1 above)
Comment (FiboApe) [reply apps/root/src/services/earnService.ts:545]: Its going to need it anyway if he wants to withdraw with it, we are saving a step by doing it here

File: apps/root/src/pages/strategy-guardian-detail/strategy-management/deposit/hooks/useEarnDepositActions.tsx (hunk 2 above)
Comment (FiboApe) [reply apps/root/src/pages/strategy-guardian-detail/strategy-management/deposit/hooks/useEarnDepositActions.tsx:612]: No need for companion permissions on a position that does not exist yet :)

Comment (th0rOdinson) [reply apps/root/src/pages/strategy-guardian-detail/strategy-management/deposit/hooks/useEarnDepositActions.tsx:612]: damn right

**Review by th0rOdinson — APPROVED** [review]

**Pull request merged**